### ISSUE TITLE
ldflag checking can result in spurious errors on the console output

### ISSUE THREAD

**Issue by electrofelix** [asottile/setuptools-golang#48]
When checking the ldflags errors may be emitted that can result in confusion for the end user
```
/usr/bin/ld: /tmp/user/16777216/cce0b3z1.o: in function `main':
test.c:(.text+0xe): undefined reference to `f'
collect2: error: ld returned 1 exit status
```

This appears to be the result of https://github.com/asottile/setuptools-golang/blob/master/setuptools_golang.py#L81 due to check_call emitting stdout/stderr by default. Perhaps these should be captured and provided with an explanation instead? https://docs.python.org/3.6/library/subprocess.html#subprocess.check_call allows for stdout/stderr to be captured or perhaps switching to check_output and pipe stderr to stdout and capture the contents of stdout to be output with context?

**Comment by asottile** [comment]
feel free to send a patch and test it on all the platforms, iirc this is needed because one of the compilers misbehaves when piped and pulling out ptys for this was severely overkill

**Comment by ankrgyl** [comment]
I just hit this on Linux (Ubuntu 20.04) as well:

```
$ cat /etc/*release*
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.5 LTS"
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

@asottile are you still open to a patch/fix?

**Comment by asottile** [comment]
see: the comment right above yours

**Comment by ankrgyl** [comment]
Yes just clarifying since it was 2 years ago :). I'll dig further, thanks for the quick reply.

**Comment by asottile** [comment]
closing this out since I'm archiving the project
